### ISSUE TITLE
Classifier OOM when computing on test set. 

### ISSUE THREAD

**Issue by AlexanderMath** [jhjacobsen/invertible-resnet#15]
Thanks for a great repository, the code works very well, is nicely documented and the overall structure is intuitive. 
I found a minor issue which can easily be solved. 

The function ```test(..)``` computes loss on test set without turning gradient computations off. 

https://github.com/jhjacobsen/invertible-resnet/blob/master/models/utils_cifar.py#L194
![image](https://user-images.githubusercontent.com/8614529/87700816-3ddb6900-c797-11ea-9161-6e5105ce0af8.png)

One might think ```model.eval()``` turns off gradients, but it does not, see e.g. [1]. 
Instead, one needs something like 

```
model.eval()
with torch.no_grad(): 
    # code from before 
```

This does usually not cause OOM, but if one is training multiple classifiers at the same time on the same GPU it does. 
This is useful when e.g. repeating experiment to get error bars). 

[1] https://discuss.pytorch.org/t/model-eval-vs-with-torch-no-grad/19615

**Comment by jhjacobsen** [comment]
Thank you, will add a fix shortly.
